libs/process_limits/Time_limit.ml

```diff
@@ -48,11 +48,26 @@ let string_of_timeout_info { Exception.name; max_duration } =
 (* could be in Control section *)
 
 (* TODO: Limit by allocations. *)
-let set_timeout (_caps : < Cap.time_limit >) ~name max_duration f =
+(* XXX: The sleeping threads delays the whole program! Hence the incremental sleeping,
+ * and the approach of stopping the sleeping thread when the work is done. Ideally,
+ * the granularity parameter should be set also by CLI flag. *)
+let set_timeout (_caps : < Cap.time_limit >) ?(granularity_float_s=0.1) ~name max_duration f =
   let token = M.Token.create () in
-  let timeout () = Thread.delay max_duration; M.Token.set token in
-  ignore (Thread.create timeout ());
-  match M.limit_with_token ~token f with
+  (* TODO:
```

Comment:
github issue?

Reply:
This one is not worth I guess, just wanted the comment to be there, will remove the TODO.

Reply:
Converted to simple comment, since the code works well already.